ath79: update qca9533 to new irq driver

lucize opened the pull request:
Commit c7efc93 renamed controller name
to qca,ar9340-intc and added some functions but qca9533.dtsi was overlooked.
Correct the dtsi and adust it to the new format

Add gmac and correct reset for cascaded irq and build-in switch
Also add the reference clock to soc dtsi so we don't have to have it in every dts

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

lucize commented:
@blogic better ?

981213 commented:
there are more things missing in the qca9533.dtsi:
cpuintc node is missing ddr-wb-channel (it should be the same as ar934x):
```
&cpuintc {
	qca,ddr-wb-channel-interrupts = <2>, <3>, <4>, <5>;
	qca,ddr-wb-channels = <&ddr_ctrl 4>, <&ddr_ctrl 2>,
				<&ddr_ctrl 0>, <&ddr_ctrl 1>;
};
```
missing gmac config node inside ahb node (it should be the same as ar933x):
```
		gmac: gmac@18070000 {
			compatible = "qca,ar9330-gmac";
			reg = <0x18070000 0x4>;
		};
```
I guess to swap phy0 and phy4 the ethernet related nodes should be:
```
&mdio1 {
	status = "okay";
	phy4-mii-enable;
	phy0: ethernet-phy@0 {
		reg = <0>;
		phy-mode = "mii";
	};
};

&eth1 {
	status = "okay";
	phy-mode = "gmii";

	gmac-config {
		device = <&gmac>;
		switch-phy-swap = <1>;
		switch-phy-addr-swap = <1>;
	};

	fixed-link {
		speed = <1000>;
		full-duplex;
	};
};

&eth0 {
	status = "okay";
	phy-mode = "mii";

	phy-handle = <&phy0>;
};
```

lucize commented:
Updated and on target all is working now.

juppin commented on target/linux/ath79/dts/qca9533.dtsi:
```diff
@@ -204,6 +213,12 @@
 
+&cpuintc {
+	qca,ddr-wb-channel-interrupts = <2>, <3>, <4>, <5>;
+	qca,ddr-wb-channels = <&ddr_ctrl 4>, <&ddr_ctrl 2>,
```
Please remove <2> from "qca,ddr-wb-channel-interrupts" and <&ddr_ctrl 4> from "qca,ddr-wb-channels" from cpuintc node because this is already handled on intc2 node.

lucize committed:
ath79: update qca9533 to new irq driver

Commit c7efc93 renamed controller name
to qca,ar9340-intc and added some functions but qca9533.dtsi was overlooked.
Correct the dtsi and adust it to the new format

Add gmac and correct reset for cascaded irq and build-in switch
Also add the reference clock to soc dtsi so we don't have to have it in every dts

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

981213 commented:
@mkresin @blogic What's the current status of this PR? I may need this for #1146

blogic commented:
merged, Thanks !

981213 commented:
Whoops! there is a mistake here: The compatible string of gmac_config should be "qca,ar9330-gmac" instead of "qca,ar9340-gmac"
And the switch reset in mdio1 was added by me in #1114 which I've dropped.

blogic commented:
dropped it from my local tree again. please open a new PR with the correct patch

lucize commented:
@981213 please correct it in your switch PR, I'm on holidays and with limited time access on computer

981213 commented:
@lucize I've done that in #1146 :)
